torch/csrc/autograd/engine.cpp

```diff
@@ -691,17 +691,29 @@ void validate_outputs(
       // AT_ERROR(format_error(ss.str()));
       continue;
     }
-    if (!grad.sizes().equals(metadata.shape())) {
-      if (!at::is_expandable_to(metadata.shape(), grad.sizes())) {
+
+    if (metadata.is_nested_tensor()) {
```

Comment:
Same as above: this should not be nested Tensor specific

Reply:
Totally agree need to be more shape agnostic